src/Components/Users/Users.tsx

```diff
@@ -19,15 +31,159 @@ export default function Users() {
     })
       .then((response) => {
         setUserList(response?.data?.data);
-        // setCurrentPage(pageNo);
       })
       .catch((error) => {
         console.log(error);
       })
-    }
+  }
+  //View Users API
+  const getUsersDetails = (id) => {
+    axios
+      .get(`${baseUrl}/Users/${id}`, {
+        headers: requestHeaders,
+      })
+
+      .then((response) => {
+        setUserDetails(response.data);
+
+      })
+      .catch((error) => {
+        console.log(error);
+      })
+  }
+  // toggle activated employee
+  const activatedEmployee = (itemId: number, data: any) => {
+    axios.put(`${baseUrl}/Users/${itemId}`, data, {
+      headers: requestHeaders,
+    })
+      .then((response) => {
+        console.log(response);
+
+        getAllUsers();
+        getToastValue("success", response?.data?.message || "Updated Sucessfully")
+      })
+      .catch((error) => {
+        console.log(error);
+
+        getToastValue("error", error?.response?.data?.message || "Error in updating the status");
+      })
+  }
+
+
+  // Toggle activation status
+  const toggleActivationStatus = (itemId: number, isActivated: boolean) => {
+    const updatedData = { isActivated: !isActivated };
+    activatedEmployee(itemId, updatedData);
+  };
+
+  //view Model
+  const showViewModel = (id: number) => {
+    setItemId(id);
+    setModelState("view-model")
+    getUsersDetails(id)
+  }
+
+  useEffect(() => {
+    getAllUsers()
+  }, [])
+
+  // Search by name
+  const getNameValue = (input) => {
+    console.log(input);
+    setSearchString(input.target.value);
+    getAllUsers(1, input.target.value);
+  }
 
   return (
-    <div>Users</div>
+    <>
+      {/* View Model */}
+      <Modal show={modelState === "view-model"} onHide={handleClose}>
+
+        <Modal.Body>
+          <h4 className=' text-whit bg-success text-center'> User Details </h4>
+          <div className='text-center'>
+            {userDetails?.imagePath ?
+              <img
+                className=' img-fluid'
+                src={`http://upskilling-egypt.com:3003` + userDetails?.imagePath} alt="" /> :
+              <img className='img-fluid' src={noData} />
+            }
+            <p><span className={`${style.title}`}>User Name:</span>{userDetails?.userName}</p>
+            <p><span className={`${style.title}`}>Email:</span>{userDetails?.email}</p>
+            <p><span className={`${style.title}`}>Phone:</span>{userDetails?.phoneNumber}</p>
+
+            <button
+              onClick={handleClose}
+              className='btn w-50'>Close
+            </button>
+          </div>
+
+        </Modal.Body>
+
+      </Modal>
+      {/* Header */}
+      <div className='header d-flex justify-content-between p-3'>
+        <h3>Users</h3>
+
+      </div>
+
+      {userList.length > 0 ?
```

Comment:
When searching for a user that doesn't exist, the search bar disappears, you should only hide the table content and replace it with NoData component but not the search bar